src/kernel/serial.zig

```diff
@@ -0,0 +1,74 @@
+const arch = @import("arch.zig").internals;
+
+const LCR = 3;
+const BAUD_MAX = 115200;
+const CHAR_LEN = 8;
+const SINGLE_STOP_BIT = true;
+const PARITY_BIT = false;
+
+pub const DEFAULT_BAUDRATE = 38400;
+
+const SerialError = error {
+    InvalidBaud
+};
+
+pub const Port = enum(u16) {
+    COM1 = 0x3F8,
+    COM2 = 0x2F8,
+    COM3 = 0x3E8,
+    COM4 = 0x2E8
+};
+
+// Compute a value that encodes the serial properties
+fn lcr_val(char_len: u8, comptime stop_bit: bool, comptime parity_bit: bool, msb: u8) u8 {
```

Comment:
Could u also expand the name of this function or abbreviated as im not sure what lcr stands for. Also camleCase.